### ISSUE TITLE
Material Design rework concept by @juliakorbut

### ISSUE THREAD

**Issue by jancborchardt** [k9mail/k-9#645]
So @cketti found the awesome concept for a K-9 rework using Material Design by @juliakorbut at http://julia.io/works/k-9/index.html
![](http://julia.io/works/k-9/images/k9_01.png)
(Only one of the screens, see the [website](http://julia.io/works/k-9/index.html) for everything.)

Except for a few things this is a great analysis and design. @juliakorbut would be great if you could chime in since you are the original author! :)
- (from @cketti) We should not be using tabs for the main folders because it would clash with using swipe for displaying the actions https://github.com/k9mail/k-9/issues/623 – the folders are good in the sidebar as shown in a later screen
- On the second screen, the actions are both on top and on the bottom – that’s a bit strange. They should only be in one place.
- The the actions on the bottom take up a lot of space on the bottom – We can just put the »New message« button in the header, next to the search icon.
- (from @cketti) I kind of like how the Gmail app puts the 'reply', 'reply all' and 'forward' actions at the end of the message. For that to be useful, we need to collapse the quoted text by default, though.
- On the compose screen, the »Sign« and »Encrypt« takes up a lot of vertical space. Maybe we should put it on the right of the subject field line as a lock icon, with a dropdown exposing these options? Otherwise the textarea is barely visible (as seen in the screenshot)

@juliakorbut @cketti what do you think?

cc @AMDG2 because you’re working on the Material Design rework https://github.com/k9mail/k-9/issues/599. And @vt0r who was involved in the discussion.

Let’s make this real! :rocket: :)

**Comment by ocdtrekkie** [comment]
Pretty sure the multiple horizontal FAB thing violates Material design concepts as presented by Google. (It's also terrifyingly bad.)

Microsoft recently commented that they had hugely strong feedback with regards to Windows 8/10 phone apps, with users vastly preferring actions on the bottom though: http://blogs.windows.com/bloggingwindows/2015/04/29/windows-10-design-getting-the-balance-right

**Comment by jancborchardt** [comment]
> multiple horizontal FAB

@ocdtrekkie what do you mean? Is it something I already mentioned?

**Comment by vt0r** [comment]
Yes, that was already covered in your initial comment. I'm not sure how relevant Windows design info is, but the bottom action buttons do seem to take over a big portion of the visible area there as-is. I think all the action buttons should go into the app bar at the top, as this prevents taking valuable screen real estate from within the message viewing area.

**Comment by ocdtrekkie** [comment]
@jancborchardt Look at the second image on the site link. The mockup presented there has four FABs across the bottom of the screen.

**Comment by jancborchardt** [comment]
Ok great! Added it to the original post.

**Comment by cketti** [comment]
I have mixed feelings about floating action buttons in general, but especially with the compose button in an email client. I mostly use my Android devices for reading email and almost never to compose messages. So the button only takes up space for me. But I also realize that usage patterns for an email client are very diverse. So my suggestion is to have an advanced setting to hide the floating action button.

Having multiple floating action buttons on a screen defeats the purpose of making the most used action very prominent. For the message screen the primary "action" is viewing the message, so ideally we give the message as much room as possible, i.e. not use a floating action button at all.
The most important actions in that screen are probably also very user-dependent. We need to come up with a reasonable default for the top 3-4 actions that will be displayed in the toolbar, the rest goes into the overflow menu. Which actions are displayed should also be configurable under advanced settings.

I kind of like how the Gmail app puts the 'reply', 'reply all' and 'forward' actions at the end of the message. For that to be useful, we need to collapse the quoted text by default, though. Otherwise the user has to scroll over all of the previous conversation.
What I don't like about the Gmail app (and to a lesser degree with Julia's design concept) is that the message header and body are visually very similar. I believe I'd prefer a slightly different background color for the message header.

I like the drawer design, especially the button to add an additional account.

All in all this is a very nice design concept. Thank you very much for putting it out there, @juliakorbut.

**Comment by jancborchardt** [comment]
> So my suggestion is to have an advanced setting to hide the floating action button.

@cketti @vt0r as long as we only have the floating action buttons as an issue we need to discuss that’s fine. :D 
I would strongly suggest to not add a setting though! (That’s probably what made K-9 so bloated in the first place. ;) We can just put the »New message« button in the header, next to the search icon.

> I kind of like how the Gmail app puts the 'reply', 'reply all' and 'forward' actions at the end of the message. For that to be useful, we need to collapse the quoted text by default, though. Otherwise the user has to scroll over all of the previous conversation.

Yeah, that we should definitely do. Not float them, but only put them at the end. And yes, of course collapse quoted text by default.

> What I don't like about the Gmail app (and to a lesser degree with Julia's design concept) is that the message header and body are visually very similar. I believe I'd prefer a slightly different background color for the message header.

I feel they are separate enough with the line and the whitespace. But yeah, we could also use a slightly grey background for the header for example. :)

**Comment by ocdtrekkie** [comment]
@jancborchardt Settings are K-9s primary feature. Without all of those settings K-9 has really no reason to exist.

**Comment by jancborchardt** [comment]
> Settings are K-9s primary feature. Without all of those settings K-9 has really no reason to exist.

@ocdtrekkie wow, I’ve now read it all.
![](https://media.giphy.com/media/zG915gt8Ok40M/giphy.gif)

And I seriously assumed that the primary features of K-9 _Mail_ were reading and writing mail instead of adjusting settings.

(Sorry but since you always seem to be one of the first to comment on new issues and demotivate new contributors – especially on the Material Design rework https://github.com/k9mail/k-9/issues/599 – I have no reason anymore to take you seriously.)

**Comment by ocdtrekkie** [comment]
If you want a badly designed open source app that just reads and writes mail, there already is one. It's called Email, from Google. 

When you pick a third party client like K-9,its because you need things different. A different UI or different settings. Why are you so committed to sabotaging every single reason someone might use this app?

**Comment by ocdtrekkie** [comment]
Perhaps a larger discussion on the email list is warranted on why people choose K-9 and what it's key differentiating features are before any major sabotage to the source is undertaken.

**Comment by jancborchardt** [comment]
@ocdtrekkie you are in the IRC channel – let’s maybe talk about this there. I feel when most people (including the project owners) think this is a good direction then we need to resolve this. (Sorry, I am not on any mailing list since I feel they only result in endless discussions, wasted time and rarely any good results.)

**Comment by cketti** [comment]
@ocdtrekkie: Your repeated destructive behavior is not welcome. Please refrain from commenting on any K-9 Mail issues on Github in the future.
If you feel the need to discuss your vision of the app, feel free to use the mailing list.

**Comment by jberkel** [comment]
Stop feeding the trolls.

**Comment by ocdtrekkie** [comment]
Banning criticism of bad ideas seems like a brilliant maintainer strategy, @cketti. Top of the class.

**Comment by dsoundmn** [comment]
I see all the points so far here. @cketti and @ocdtrekkie have valid sides on both points. I do use K9 because of a number of reasons 1) it has the features I enjoy and supports more standards than other apps (even to this point in time) 2) I am able to get to my email quickly and read quickly and 3) The ability to adjust settings at a very granular level. I use 2 more than anything else and would see that as likely most important.

I have been using the dev builds for quite some time and like what I'm seeing so far in improvements in reading emails and not having white screens.... so first, Thank you.

The point related to the other apps that have a better interface at this time (inbox / email / gmail) etc... all of which now have better support than when I first used K9, but not up to the class of K9 imo. The interface for me needs work as it relates to newer designs and simpler functionality - the most common for me is triggering multi-select. I find it to not be as simple as other apps and would not mind a bit if it worked slightly better. I like some of what outlook's app has added in this functionality. 

What if you setup some way to survey k9 users as to what they like and don't like about it? Say using a tiny one time pop up or other thing in the next version.

**Comment by frewsxcv** [comment]
Wow, that concept art looks great! I can't wait to see this go live :sparkles:

**Comment by Natanji** [comment]
I would like to chime in and add the ability to use left/right swipe on single mesages like in Google's Inbox. The actions could be freely configurable (the Boxer mail app already has a similar feature). It would enable users to e.g. define an "archive" action by swiping a message to the left, thereby allowing them the "Inbox Zero" approach that Google is now introducing with their Inbox app.

Is there any chance we will see a similar support in a material design of K-9?

**Comment by jancborchardt** [comment]
@Natanji does Google Inbox use left-right swiping on single messages for these actions? Normally left-right swiping is used for moving to the next/previous message.

**Comment by Natanji** [comment]
To clarify: when the list of messages is displayed, left/right swipe on a
message can do those actions. In the message view (after opening it), those
actions are instead located on the top side of the message.

Jan-Christoph Borchardt <notifications@github.com> schrieb am Mo., 6. Juli
2015 17:05:

**Comment by jancborchardt** [comment]
Ah, sure. Yeah, that’s standard by now.

**Comment by Manko10** [comment]
I just checked out the current material_design branch. Right now it looks like a hybrid between old Holo K9 and something that has something to do with Material. Since it's clearly work in progress, I would like to ask how much of this draft is planned to be integrated into the new design. I really love the draft above and would be very happy to see such a beautiful and simple interface in the future.
I always found the many different screens for accounts, folder, messages and conversations to be really confusing and difficult to use. And although Holo is already a lot more modern than the previous Gingerbread design, it clearly looks dated.

/cc issue #599

**Comment by JasperWeiss** [comment]
The reason why I would use k-9 mail is A, it's opensource and available on f-droid, B it supports Openkeychain (brilliant app btw)
The reason why I don't use it is because of the ancient design. 
Contrary to what mr 'screw material design' said, the horizontal fab is very material designish. It's used in Youtube and whatsapp aswell. Overall I feel everyone (except for some people who aren't willing to move on) want k-9 mail to adopt material design. 
It shouldn't be hard to have a option in settings that enables the obsolete holo design though. If not they can always compile it themselves.

As soon as k-9 is material design, amaze file manager supports sftp, freeOtp is material design and supports backups, Moxie finally aggrees to distributing signal on f-droid. I can attempt to go Google free.  (And firefox gets updates on f-droid again)

**Comment by ocdtrekkie** [comment]
@JasperWeiss You can't go "Google-free" while insisting every app you use be plastered with Google's proprietary branding. Unfortunately, you're far more attached to them than you even realize.

**Comment by JasperWeiss** [comment]
@ocdtrekkie Google is one of the primary contributors to android so no indeed I can not go Google free. I should rephrase that as play store/services free. I like what Google does, it pushes technology forward, offers great services. I just dislike their data mining and proprietary play store being installed as a privilleged app that has full system access.

I'm not sure what you're trying to archieve with all the hate on the new design, either way it's going to be implemented. Please tell me, I'd like to now why you dislike it so much. Would an option for holo design not solve your complaints?

**Comment by ocdtrekkie** [comment]
@JasperWeiss An option to support Holo would, but it's also a mostly unreasonable thing to maintain. Holo and Material are different to the level that supporting both UIs within a single app would be extremely burdensome. It'd be better/more preferable to leave the mail app millions of us use and rely on daily intact, and create a separate app for users who want something so significantly different. 

As to specific reasons for disliking Material, they're numerous, primarily involving design patterns which are less efficient and distinctly impair productivity.

**Comment by JasperWeiss** [comment]
All right, what about we make a fork with material design. That way we can change the icon to something more material aswell. Everybody is happy without compromises.
This way we can release it much quicker because we can release the fork as a beta.

**Comment by cketti** [comment]
@JasperWeiss @ocdtrekkie: Please take your argument elsewhere. The issue tracker is the wrong place for that.

**Comment by frewsxcv** [comment]
Going back on topic: What is the status of this? What are the next steps to be done?

**Comment by JasperWeiss** [comment]
@cketti Sorry. This is an issue though. Either way I'm perfectly happy but I'm just trying to make everybody happy with the direction k-9 mail is going.

**Comment by ocdtrekkie** [comment]
@JasperWeiss Yes, that would be a far superior way to handle it for everyone involved. My largest concern is that millions of users auto-update their apps, and will end up getting a massively less ideal experience, and no way to revert it, because unlike F-Droid, the Play Store doesn't allow users access to previous versions, and the whole Android platform offers no data-preserving downgrade method. Ideally, such a radical shift should be released as a separate app, and then the original app should recommend it/let people know the new app is an available option if they'd prefer.

Sorry @cketti, I respond when commented about. And I even think Jasper and I have come to some sort of understanding of our respective concerns.

**Comment by cketti** [comment]
@frewsxcv: See #599

@JasperWeiss: You're not the one deciding on the direction K-9 Mail is going :)

**Comment by JasperWeiss** [comment]
@cketti I never said it is my call to make. Again I'm really appreciative about everything you guys are doing and I'm perfectly happy either way. I was just suggesting a course of action that would address the valid concerns of @ocdtrekkie

**Comment by Valodim** [comment]
I would argue that the fork should be the classic holo version, not the other way around. Any way you slice it, people who don't like material design are the minority.

**Comment by JasperWeiss** [comment]
@Valodim I think most users would be happy with a material design overhaul. We could make a 'classic' fork for those who don't. Either way we shouldn't just dismiss the minority don't you think? Personally I don't see how one could like holo over material but it wouldn't hurt to have it as an option.

**Comment by jancborchardt** [comment]
Can we please stop this banter about _if_ it is a good idea to move to Material design? The longer this discussion goes on, the more people will move to better & modern looking email apps.

> it wouldn't hurt to have it as an option

@JasperWeiss yes it would. It would take development resources and focus away from the proper version. If @ocdtrekkie is so keen on Holo, he can fork and maintain it – although it will of course be too much work.
The real maintainers (e.g. @cketti) clearly want to go the proper Material Design route and this discussion is simply frustrating.

**Comment by Manko10** [comment]
Thanks Jan. This discussion is really getting out of hand. I don't get
the point why there has to be an alternative Holo design. The Android
platform is moving forward and apps have to follow it. It's as simple as
that.
But maybe there are also some Gingerbread users who like the old
Gingerbread design better. Maybe we should add an option for them, too.
Any volunteers? No? Disappointing.

**Comment by JasperWeiss** [comment]
@jancborchardt I aggree. Let me be clear that I want material design. I was just proposing a solution for @ocdtrekkie. By all means, forget about holo and make this happen. He can indeed compile it himself as most people (everyone but some) want material design.

**Comment by JasperWeiss** [comment]
So how far are you guys with material design?

**Comment by Valodim** [comment]
As far as I know, nobody started yet.

**Comment by jancborchardt** [comment]
Well, @AMDG2 started in #599. And I guess he appreciates any help. :)

**Comment by Valodim** [comment]
Not to sound dismissive, but the single commit in his branch replaces a bunch of icons with material icons. My point stands.

**Comment by jancborchardt** [comment]
Might be due to the quite negative first response in his pull request, unfortunately …

**Comment by AMDG2** [comment]
Negative feedback one thing. But time is another. 
I would like to continue my embryon of work. But due to lack a free time since I started I may not. 
I had plenty of free time, but other projects in parallel. Since this fork is getting negative feedback and is generating a lot of useless chatting (because some persons are always negative) I didn't make this a first one priority. 

@cketti thank you for your great work ;) 

Le 12 janvier 2016 16:54:35 GMT+01:00, Jan-Christoph Borchardt <notifications@github.com> a écrit :
>Might be due to the quite negative first response in his pull request,
>unfortunately …
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/k9mail/k-9/issues/645#issuecomment-170955267

-- 
Envoyé de mon appareil Android avec K-9 Mail. Veuillez excuser ma brièveté.

**Comment by jancborchardt** [comment]
@AMDG2 rest assured that the majority (most importantly, the actual maintainers of the app) really likes your work, and wants to move forward to Material Design. :)

**Comment by JasperWeiss** [comment]
I'm really looking forward to a design overhaul

**Comment by alexara** [comment]
Are there any news on this? It's now a year since this thread has been opened and I would really welcome to see your design or at least parts of it in a future release of K-9 mail. :-)

**Comment by baconicsynergy** [comment]
Me too! I would love to see this implemented. Perhaps an icon makeover too.

**Comment by HammyHavoc** [comment]
Yeah, I personally hate the icon for K-9 as I'm now ashamed to admit to having ever been a Whovian, I also think K-9's current design is cumbersome and not very easy on the eyes, which is a pity given how much potential K-9 has.

**Comment by juliakorbut** [comment]
Hey all! Sorry for not being fully responsive in the past. In case anyone is looking at this thread, you can find an updated, annotated, and full version of the designs here: 
https://github.com/juliakorbut/k-9-design/ 

Feedback is always welcome. Thank you!

**Comment by juliakorbut** [comment]
Hey all! Sorry for not being fully responsive in the past. In case anyone is looking at this thread, you can find an updated, annotated, and full version of the designs here:
https://github.com/juliakorbut/k-9-design/

Feedback is always welcome. Thank you! :)

**Comment by GuillaumeAmat** [comment]
I love it!

**Comment by 12people** [comment]
@juliakorbut I've downloaded the Sketch files, but I have trouble editing them (the artboards appear to be empty). How can I modify them?

**Comment by juliakorbut** [comment]
@12people Odd. It works fine for me. Have you tried zooming to canvas? _(View > Center Canvas)_

**Comment by juliakorbut** [comment]
@12people @12people Odd. It works fine for me. Have you tried zooming to canvas? _(View > Center Canvas)_

For annotations, the Specs folder or Zeplin link might be better. With a live Sketch file there's a risk of moving layers around

**Comment by 12people** [comment]
@juliakorbut I can see the artboards and their contents just fine, but I can't edit them. In the panel on the left, the artboards appear to have no contents.

**Comment by juliakorbut** [comment]
@12people Could be that the layers are toggled off - can you try clicking here?

![togglelayers](https://cloud.githubusercontent.com/assets/8000951/23332658/50638a42-fb4b-11e6-9c69-af91d0201dfa.png)

**Comment by juliakorbut** [comment]
Does anyone have suggestions on ways to structure this into a proper project, with issues tracking? 

To me it would make sense to open an entirely new milestone, make a project per screen, and assign an issue per every feature in those screens. It would make it easier for me as a designer to address things on a feature level and work in parallel with everyone. Kind of like having design sub-tasks for user stories in JIRA.

**Comment by juliakorbut** [comment]
Does anyone have suggestions on ways to structure this into a proper project, with issues tracking? 

To me it would make sense to open an entirely new milestone, make a project per screen, and assign an issue per every feature in those screens. It would make it easier for me as a designer to address things on a feature level and work in parallel with everyone. Kind of like having design sub-tasks for user stories in JIRA.

For example:

## K-9 2017 Redesign (milestone)

**Drawer (project)**
**Settings (project)**
**Folder view (project)**
>Swipe to delete (issue)
>Swipe to archive (issue)
>Multi select (issue)
>etc etc

**Message view (project)**
**Compose (project)**

**Comment by philipwhiuk** [comment]
I don't know that forking the project (so k9_materialChanges as a separate project under k9mail) worked that well for the PGP stuff but that would give the structure you want.

This probably needs some discussion.

**Comment by cketti** [comment]
Material Design rework is underway.

**Comment by jancborchardt** [comment]
Woah @cketti congrats! :) Let me know if there’s any input needed, we could also meet again for a design session.
